test/gurka/test_construction.cc

```diff
@@ -0,0 +1,159 @@
+#include "gurka.h"
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+class IncludeConstructionTest : public ::testing::Test {
+protected:
+  static gurka::nodelayout layout;
+  static gurka::ways ways;
+
+  static void SetUpTestSuite() {
+    const std::string ascii_map = R"(
+      A---------B--------C
+                         |
+                         D
+    )";
+
+    ways = {
+        {"AB", {{"highway", "primary"}}},
+        {"BC", {{"highway", "construction"}, {"construction", "primary"}}},
+        // road under construction with missing 'construction' tag; should be excluded from the graph
+        {"CD", {{"highway", "construction"}}},
+    };
+
+    layout = gurka::detail::map_to_coordinates(ascii_map, 100);
+  }
+};
+
+gurka::nodelayout IncludeConstructionTest::layout = {};
+gurka::ways IncludeConstructionTest::ways = {};
+
+TEST_F(IncludeConstructionTest, CheckConstructionsExcluded) {
+  // exclude roads under construction
+  auto map = gurka::buildtiles(layout, ways, {}, {}, "test/data/construction",
+                               {{"mjolnir.include_constructions", "false"}});
+
+  baldr::GraphReader reader(map.config.get_child("mjolnir"));
+
+  auto AB = gurka::findEdge(reader, layout, "AB", "B");
+  ASSERT_NE(std::get<0>(AB), baldr::GraphId{}) << "missing valid road AB";
+
+  auto BC = gurka::findEdge(reader, layout, "BC", "C");
+  EXPECT_EQ(std::get<0>(BC), baldr::GraphId{}) << "constructions should be excluded";
```

Comment:
`include_constructions=false` -> it's expected that constructions will be excluded from the graph